torchvision/transforms/v2/functional/_color.py

```diff
@@ -65,6 +65,30 @@ def _rgb_to_grayscale_image_pil(image: PIL.Image.Image, num_output_channels: int
     return _FP.to_grayscale(image, num_output_channels=num_output_channels)
 
 
+def grayscale_to_rgb(inpt: torch.Tensor) -> torch.Tensor:
+    """See :class:`~torchvision.transforms.v2.GrayscaleToRgb` for details."""
+    if torch.jit.is_scripting():
+        return grayscale_to_rgb_image(inpt)
+
+    _log_api_usage_once(grayscale_to_rgb)
+
+    kernel = _get_kernel(grayscale_to_rgb, type(inpt))
+    return kernel(inpt)
+
+
+@_register_kernel_internal(grayscale_to_rgb, torch.Tensor)
+@_register_kernel_internal(grayscale_to_rgb, tv_tensors.Image)
+def grayscale_to_rgb_image(image: torch.Tensor) -> torch.Tensor:
+    # rgb_to_grayscale can be used to add channels so we reuse that function.
+    return _rgb_to_grayscale_image(image, num_output_channels=3, preserve_dtype=True)
+
+
+@_register_kernel_internal(grayscale_to_rgb, PIL.Image.Image)
+def grayscale_to_rgb_image_pil(image: PIL.Image.Image) -> PIL.Image.Image:
+    # to_grayscale can expand channels from 1 to 3 so we reuse that function.
+    return _FP.to_grayscale(image, num_output_channels=3)
```

Comment:
Calling `*to_grayscale` will properly make the image 3D, but it will also convert it to grayscale which is not desirable (for both tensors and PIL images). Take a look at the output of this

```py
# %%
import torch
from torchvision.transforms.v2.functional import grayscale_to_rgb, to_pil_image

#%%
img = torch.rand(3, 300, 400)  # RGB
pil_img = to_pil_image(img)  # RGB

# %%
pil_img
# %%
to_pil_image(grayscale_to_rgb(img))  # this is grayscale

# %%

grayscale_to_rgb(to_pil_image(img))  # This is grayscale
```

(open it in vscode to make it a notebook) and you should see that the images are converted to grayscale.

Reply:
Good catch. I addressed this issue for Tensors and for PILs, now I am simply calling the convert function which deals with this by itself. Also added a test for it.